src/user/user.service.ts

```diff
@@ -1,29 +1,146 @@
-import { Injectable, NotFoundException } from '@nestjs/common';
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
-import { User } from './entities/user.entity';
-import { Repository } from 'typeorm';
-import { UserDto } from './dto/user';
+import { UNIQUE_USER_EMAIL_CONSTRAINT, User } from './entities/user.entity';
+import { QueryFailedError, Repository } from 'typeorm';
+import { ActivateUserDto, CreateUserDto, UserDto } from './dto/user';
+import { UserAlreadyExists, UserNotActivated } from './user.exceptions';
+import { CryptoService } from '../utils/crypto.service';
+import { UserToken, UserTokenType } from './entities/user-token.entity';
 
 @Injectable()
 export class UserService {
   constructor(
     @InjectRepository(User)
     private readonly userRepository: Repository<User>,
+    @InjectRepository(UserToken)
+    private readonly tokenRepository: Repository<UserToken>,
+    private readonly cryptoService: CryptoService,
   ) {}
 
   async findAll(): Promise<UserDto[]> {
+    //Todo: do we need this route? Filter active/inactive?
     return this.userRepository.find();
   }
 
-  async findOne(username: string): Promise<UserDto> {
+  /**
+   * Returns on user or throws a NotFoundException. If canBeInactive is set, it
+   * also returns a user that has not activated their account yet. Defaults to
+   * false so we always scope this to the active users only and throw a
+   * UserNotActivatedException.
+   * @param email
+   * @param canBeInactive
+   */
+  async findOne(email: string, canBeInactive = false): Promise<UserDto> {
```

Comment:
I propose to rename "canBeInactive" to "showInactiveUsers". It makes it more clear what the parameter is about in my opinion.

Reply:
I would argue otherwise - because you want to find a user, you submit selection criteria, which are, in this case, only the `email`  and the activation state. As such, I feel that `canBeInactive` is more accurate because its a criterion the user has or does not, where as `showInactive` would be from an application perspective.